python/sparkdl/graph/utils.py

```diff
@@ -0,0 +1,255 @@
+
+The various combination makes it hard to figuring out the best way.
+We provide some methods to map whatever we have as input to
+one of the four target variants.
+"""
+
+def validated_graph(graph):
+    """
+    Check if the input is a valid tf.Graph
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    """
+    assert isinstance(graph, tf.Graph), 'must provide tf.Graph, but get {}'.format(type(graph))
+    return graph
+
+def get_shape(graph, tfobj_or_name):
+    """
+    Return the shape of the tensor as a list
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
+    """
+    graph = validated_graph(graph)
+    _shape = get_tensor(graph, tfobj_or_name).get_shape().as_list()
+    return [-1 if x is None else x for x in _shape]
+
+def get_op(graph, tfobj_or_name):
+    """
+    Get a tf.Operation object
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
+    """
+    graph = validated_graph(graph)
+    if isinstance(tfobj_or_name, tf.Operation):
+        return tfobj_or_name
+    name = tfobj_or_name
+    if isinstance(tfobj_or_name, tf.Tensor):
+        name = tfobj_or_name.name
+    if not isinstance(name, six.string_types):
+        raise TypeError('invalid op request for {} of {}'.format(name, type(name)))
+    _op_name = as_op_name(name)
+    op = graph.get_operation_by_name(_op_name)
+    assert op is not None, \
+        'cannot locate op {} in current graph'.format(_op_name)
+    return op
+
+def get_tensor(graph, tfobj_or_name):
+    """
+    Get a tf.Tensor object
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
+    """
+    graph = validated_graph(graph)
+    if isinstance(tfobj_or_name, tf.Tensor):
+        return tfobj_or_name
+    name = tfobj_or_name
+    if isinstance(tfobj_or_name, tf.Operation):
+        name = tfobj_or_name.name
+    if not isinstance(name, six.string_types):
+        raise TypeError('invalid tensor request for {} of {}'.format(name, type(name)))
+    _tensor_name = as_tensor_name(name)
+    tnsr = graph.get_tensor_by_name(_tensor_name)
+    assert tnsr is not None, \
+        'cannot locate tensor {} in current graph'.format(_tensor_name)
+    return tnsr
+
+def as_tensor_name(name):
+    """
+    Derive tf.Tensor name from an op/tensor name
+
+    :param name: op name or tensor name
+    """
+    assert isinstance(name, six.string_types)
+    name_parts = name.split(":")
+    assert len(name_parts) <= 2
+    if len(name_parts) < 2:
+        name += ":0"
+    return name
+
+def as_op_name(name):
+    """
+    Derive tf.Operation name from an op/tensor name
+
+    :param name: op name or tensor name
+    """
+    assert isinstance(name, six.string_types)
+    name_parts = name.split(":")
+    assert len(name_parts) <= 2
```

Comment:
` assert len(name_parts) <= 2, name_parts` to print what is wrong when the assert gets triggered